Mark some CSS properties as first available in Safari 5 / 5.1

foolip opened the pull request:
This is based on results collected from Safari 4, 5 and 5.1 (and more):
https://github.com/foolip/mdn-bcd-results/tree/6288e5f0813157d70821c635fe05bb469ee96e0d

There are no release notes for Safari 5 or 5.1, but these changes are
likely to be correct. Tests on the form
`'propertyName' in document.body.style` were used to determine support.

The script to update results based on resutls is in development:
https://gist.github.com/foolip/bdf30013e20342b35cb54d6f5f589372

These changes were hand picked from the output as good changes.

Elchi3 commented:
So, caniuse agrees with "6" for unprefixed border-image-* support. https://caniuse.com/#feat=border-image
I tried to blame and see how it was added to caniuse, but it seems like it was in the initial commit of the repo.
In Saucelabs, I can only go back to Safari 8. Any ideas how to find if 5 or 6 is right? I don't have Apple devices.

foolip commented:
BrowserStack has Safari back to version 4. However, that is the source of this data, better to use a non-trivial test case to confirm support.

foolip commented:
(I'll find a test case and confirm.)

foolip commented:
I've tested http://software.hixie.ch/utilities/js/live-dom-viewer/saved/6812 in Safari 5.1 and 6, and judging by that it looks like it only began really working in Safari 6. Support for parsing the properties was added before the behavior was fixed.

@Elchi3 since it's now happened with both Edge (https://github.com/mdn/browser-compat-data/pull/3667#discussion_r269210818) and Safari I suspect this will be a recurring problem if trying to sync BCD's CSS data with what can be observed from simple feature detects.

I could add notes here like in https://github.com/mdn/browser-compat-data/pull/3667, but given that the problem only affects Safari 5.1 I can't imagine that would be useful information for web developers, presenting it is likely net negative even if it's accurate. How would you like to handle cases like this? Would a `maintainer_notes` field for stuff like this be a good idea?

Elchi3 commented:
I'm not sure if I like `maintainer_notes`. I will think about it. I think the question is whether BCD should store a note when information sources are wrong, or whether the information sources store that themselves as a black list. Or, ideally, the info sources are fixed so they don't propose they wrong info anymore.

foolip commented:
@Elchi3 for now I added a mechanism alongside my in-progress update script to override data known to be incorrect: https://github.com/foolip/mdn-bcd-results/commit/f70a9d4e4fc4accea52df327b81c982da6285cf4

And I've reverted the border-image-* changes on this PR, leaving only 3 changes.
